x-pack/plugins/ml/public/application/components/job_selector/job_selector_flyout.tsx

```diff
@@ -0,0 +1,289 @@
+  dateFormatTz: string;
+  selectedIds?: string[];
+  newSelection?: string[];
+  onFlyoutClose: () => void;
+  onJobsFetched?: (maps: JobSelectionMaps) => void;
+  onSelectionChange?: (newSelection: string[]) => void;
+  onSelectionConfirmed: (payload: {
+    newSelection: string[];
+    jobIds: string[];
+    groups: Array<{ groupId: string; jobIds: string[] }>;
+    time: any;
+  }) => void;
+  singleSelection: boolean;
+  timeseriesOnly: boolean;
+  maps: JobSelectionMaps;
+  withTimeRangeSelector?: boolean;
+}
+
+export const JobSelectorFlyout: FC<JobSelectorFlyoutProps> = ({
+  dateFormatTz,
+  selectedIds = [],
+  singleSelection,
+  timeseriesOnly,
+  onJobsFetched,
+  onSelectionChange,
+  onSelectionConfirmed,
+  onFlyoutClose,
+  maps,
+  withTimeRangeSelector = true,
+}) => {
+  const {
+    services: { notifications },
+  } = useMlKibana();
+
+  const [newSelection, setNewSelection] = useState(selectedIds);
+
+  const [showAllBadges, setShowAllBadges] = useState(false);
+  const [applyTimeRange, setApplyTimeRange] = useState(true);
+  const [jobs, setJobs] = useState<MlJobWithTimeRange[]>([]);
+  const [groups, setGroups] = useState<any[]>([]);
+  const [ganttBarWidth, setGanttBarWidth] = useState(DEFAULT_GANTT_BAR_WIDTH);
+  const [jobGroupsMaps, setJobGroupsMaps] = useState(maps);
+
+  const flyoutEl = useRef<{ flyout: HTMLElement }>(null);
+
+  function applySelection() {
+    // allNewSelection will be a list of all job ids (including those from groups) selected from the table
+    const allNewSelection: string[] = [];
+    const groupSelection: Array<{ groupId: string; jobIds: string[] }> = [];
+
+    newSelection.forEach(id => {
+      if (jobGroupsMaps.groupsMap[id] !== undefined) {
+        // Push all jobs from selected groups into the newSelection list
+        allNewSelection.push(...jobGroupsMaps.groupsMap[id]);
+        // if it's a group - push group obj to set in global state
+        groupSelection.push({ groupId: id, jobIds: jobGroupsMaps.groupsMap[id] });
+      } else {
+        allNewSelection.push(id);
+      }
+    });
+    // create a Set to remove duplicate values
+    const allNewSelectionUnique = Array.from(new Set(allNewSelection));
+
+    const time = applyTimeRange
+      ? getTimeRangeFromSelection(jobs, allNewSelectionUnique)
+      : undefined;
+
+    onSelectionConfirmed({
+      newSelection: allNewSelectionUnique,
+      jobIds: allNewSelectionUnique,
+      groups: groupSelection,
+      time,
+    });
+  }
+
+  function removeId(id: string) {
+    setNewSelection(newSelection.filter(item => item !== id));
+  }
+
+  function toggleTimerangeSwitch() {
+    setApplyTimeRange(!applyTimeRange);
+  }
+
+  function clearSelection() {
+    setNewSelection([]);
+  }
+
+  function handleNewSelection({ selectionFromTable }: { selectionFromTable: any }) {
+    setNewSelection(selectionFromTable);
+  }
+
+  // Wrap handleResize in useCallback as it is a dependency for useEffect on line 131 below.
```

Comment:
`on line 131` looks like it needs to be updated. Probably good enough to say `further down`?

Reply:
I just moved this code, so didn't touch any comments 🙂 I believe it worth to refactor and use `EuiResizeObserver`, will keep it for the follow-up